src/components/navigation/document-view.tsx

```diff
@@ -0,0 +1,270 @@
+      types.forEach((type) => {
+        const docsByTypeArr = documents.byType(type);
+        docsByTypeArr.forEach((doc: any) => {
+          docs.push(doc);
+        });
+      });
+    const starredDocs = docs.filter((doc: DocumentModelType) => !doc.getProperty("isDeleted") && doc.isStarred);
+    return starredDocs;
+  };
+  const starredDocuments = getStarredDocuments(documentTypes);
+  const numStarredDocs = starredDocuments.length;
+  const currentOpenDocIndex = openDocument && starredDocuments.indexOf(openDocument);
+  const sectionClass = openDocument?.type === "learningLog" ? "learning-log" : "";
+
+  function handleEditClick(document: DocumentModelType) {
+    ui.problemWorkspace.setPrimaryDocument(document);
+  }
+  // TODO: this edit button is confusing when the history is being viewed. It
+  // opens the original document for editing, not some old version of the
+  // document they might be looking at. Previously this edit button was disabled
+  // when the history document was being shown because SectionDocumentOrBrowser
+  // knew the state of playback controls. It no longer knows that state, so now
+  // the edit button is shown all of the time.
+  // PT Story: https://www.pivotaltracker.com/story/show/183416176
+  const editButton = (type: string, sClass: string, document: DocumentModelType) => {
+    return (
+      (type === "my-work") || (type === "learningLog")
+        ?
+          <div className={`edit-button ${sClass}`}
+                onClick={() => handleEditClick(document)}>
+            <EditIcon className={`edit-icon ${sClass}`} />
+            <div>Edit</div>
+          </div>
+        : null
+    );
+  };
+
+  const loadDocumentContent = async (document: DocumentModelType) => {
+    await document.fetchRemoteContent(queryClient, context);
+  };
+
+  const handleSelectDocument = (document: DocumentModelType) => {
+    if (ui.focusDocument === document.key) {
+      ui.closeSubTabDocument(tabSpec.tab, subTab.label);
+    } else {
+      if (!document.hasContent && document.isRemote) {
+        loadDocumentContent(document);
+      }
+      ui.openSubTabDocument(tabSpec.tab, subTab.label, document.key);
+      const logEvent = document.isRemote
+        ? LogEventName.VIEW_SHOW_TEACHER_NETWORK_COMPARISON_DOCUMENT
+        : LogEventName.VIEW_SHOW_COMPARISON_DOCUMENT;
+      logDocumentEvent(logEvent, { document });
+    }
+  };
+
+  // Published documents are listed in reverse order of index [n..1] so previous and next toggles are also reversed
+  // Workspaces Starred tab show the problem document as the first document in the list
+  // and personal documents are shown in reverse order. So when we get the list of starred
+  // documents, the sequence is [0] is the problem document [1..n] are the personal documents.
+  // But when displayed, the sequence is [0, n..1].
+  // So we still want to display the previous arrow when we reach n to flip to the problem
+  // document.
```

Comment:
Given that `handleChangeDocument` works the same for all document types, I think it would make sense to move this comment above `hideLeftFlipper`, where different document types are handled in different ways.